waiter/src/waiter/auth/authentication.clj

```diff
@@ -379,16 +380,36 @@
       :else
       (on-auth-required))))
 
-(defn wrap-auth-bypass
-  "Middleware that checks if authentication is disabled for a token and sets the
-  :skip-authentication key of the request to true before passing to next handler."
-  [handler]
+(defn- make-wrap-auth-bypass
+  "Takes a handler and a function to call when there is an error with processing the
+  authentication parameter. If successful and authentication is disabled for the token,
+  then the :skip-authentication is set to true for the request object before getting passed
+  to the handler"
+  [handler on-error]
   (fn auth-bypass-handler [{:keys [waiter-discovery] :as request}]
     (process-authentication-parameter
       waiter-discovery
-      (fn [status message]
-        (utils/clj->json-response {:error message} :status status))
+      (partial on-error request)
```

Comment:
Avoid using partial function. Previous experimentation has found them to have a performance penalty. We avoid including them in potential hot paths. Please replace with a `fn` that lists out the parameters.